docs/design/authentication_policy.md

```diff
@@ -0,0 +1,40 @@
+# Envoy Gateway Authentication Policy API
+
+## Overview
+
+This authen policy is to declare the authentication mechanisms, to be enforce on connection and request going though Envoy Gateway. This includes the credential (X.509, JWT, etc), parameters (cipher suites, key algorithms)
+The policy is similar to [OpenAPI 3.1 security objects](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.0.md#securitySchemeObject) without the API key part, and should be easily translatable from it with some additions.
+
+## Authentication mechanisms
+This policy should support the following authentication mechanisms:
+- JWT Bearer Token
+- mutualTLS (client certificate)
+- OAuth2
+- OIDC
+- External authentication
+
+In the first phase, Envoy Gateway will implement JWT Bearer Token authentication.
+
+In general those policy translates into Envoy HTTP filters at HTTP connection manager level, and route specific settings will be applied for each route. These APIs are expressed in a Policy CRD and attached to Gateway API resources with [Policy Attachement](https://gateway-api.sigs.k8s.io/references/policy-attachment/).
+
+## JWT Bearer Token
+
+A JWT Bearer Token authentication policy will look like the following:
+
+```
+apiVersion: gateway.envoyproxy.io/v1alpha1
+kind: Authentication
+metadata:
+  name: productpage
+spec:
+  type: jwt
+  jwt:
+    iss: https://www.okta.com
+    aud: bookinfo.com
+    jwksUri: https://bookinfo.com/jwt/public-key/jwks.json
+  targetRef:
+    kind: HTTPRoute
```

Comment:
will Gateway `targetRef` be supported ?

Reply:
yeah agreed, it would be good to clarify what resources will be allowed to be targeted by the `Authentication` Policy and further, what sort of override/defaulting behavior should be expected, re this section of the reference: https://gateway-api.sigs.k8s.io/references/policy-attachment/#hierarchy

Reply:
Why is a mutual binding required? A user should be able to create the ns-scoped authentication resource specified by this PR that describes details, e.g. type of authentication. Any resource in the ns that supports `kind: OurAuthKind`, e.g. HTTPRoute, should be able to reference it.

Reply:
It is not a mutual binding, this is a Policy Attachement and not referenced from HTTPRoute or Gateway.

Reply:
Thanks for the clarification. Based on this draft [doc](https://docs.google.com/document/d/1p_kmAOmqebX8zyHnXW5I58cD8ZhNX4QNSa_gw8uAYxw/edit?usp=sharing) from @youngnick, the policy object should not refer to a specific instance of a supported kind. Additionally, the Policy Attachment should be used to set defaults and overrides and not attaching policy.

Reply:
Considering this resolved since we agreed to take a route filter approach for authN.